backend/matching-service/controllers/matchingController.ts

```diff
@@ -0,0 +1,138 @@
+// matchingController.ts
+import { Request, Response, NextFunction } from "express";
+import matchingRequestModel from "../models/MatchingRequest";
+import { getSocket } from "../utils/socket";
+import { SOCKET_EVENTS } from "constants/socketEventNames";
+import { MATCHING_STATUS } from "constants/matchingStatus";
+
+export async function getMatch(
+    request: Request,
+    response: Response,
+    next: NextFunction
+) {
+    try {
+        const io = getSocket();
+
+        const { username, email, category, difficulty, socketId } = request.body;
+
+        const newRequest = new matchingRequestModel({
+            username,
+            email,
+            category,
+            difficulty,
+            socketId
+        });
+
+        await newRequest.save();
+
+        // Try to find a match in MongoDB based on category and difficulty
+        const match = await findMatch(category, difficulty, username);
+
+        if (match) {
+            io.to(match.socketId).emit(SOCKET_EVENTS.MATCH_FOUND, {
+                message: `You have been matched with ${username}`,
+                category,
+                difficulty,
+            });
+
+            io.to(socketId).emit(SOCKET_EVENTS.MATCH_FOUND, {
+                message: `You have been matched with ${match.username}`,
+                category,
+                difficulty,
+            });
+
+            // Remove matched users from MongoDB
+            await matchingRequestModel.findByIdAndDelete(match._id);
+            await matchingRequestModel.findByIdAndDelete(newRequest._id);
+
+            return response.status(200).json({ message: 'Match found!', matchingStatus: MATCHING_STATUS.SUCCESS });
+        } else {
+            return response.status(200).json({ message: 'Searching for a match...', matchingStatus: MATCHING_STATUS.SEARCHING });
+        }
+    } catch (error) {
+        next(error);
+    }
+}
```

Comment:
By using the pub sub model, or event driven system. you can implement a queue system without a persistance mongodb record.
Mongodb will be very slow in the real world scenario for the match system because of the write operation.

Reply:
you can trigger an match found by consume the msg

Reply:
Done